.github/workflows/javascript.yml

```diff
@@ -0,0 +1,98 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2023 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: JavaScript Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
```

Comment:
Is there a reason it shouldn't run on other branches?

For the other CI checks that we already have on GitHub Actions we use `on: [push, pull_request]` and I am wondering what is better.

Reply:
There is no specific reason for the same, but just the fact that being explicit above the branches that you want to run the workflow on usually helps limit the runs. It'd probably be better to standardize this workflow following others and run it on all PRs.